Microsoft.Toolkit.Uwp.UI.Controls/Carousel/CarouselItemAutomationPeer.cs

```diff
@@ -0,0 +1,145 @@
+        /// <returns>The UI Automation provider.</returns>
+        public IRawElementProviderSimple SelectionContainer
+        {
+            get
+            {
+                Carousel parent = this.OwnerCarouselItem.ParentCarousel;
+                if (parent == null)
+                {
+                    return null;
+                }
+
+                AutomationPeer peer = FromElement(parent);
+                return peer != null ? this.ProviderFromPeer(peer) : null;
+            }
+        }
+
+        private CarouselItem OwnerCarouselItem
+        {
+            get { return this.Owner as CarouselItem; }
+        }
+
+        /// <summary>Adds the current element to the collection of selected items.</summary>
+        public void AddToSelection()
+        {
+            CarouselItem owner = this.OwnerCarouselItem;
+            Carousel parent = owner.ParentCarousel;
+            parent.SetSelectedItem(owner);
+        }
+
+        /// <summary>Removes the current element from the collection of selected items.</summary>
+        public void RemoveFromSelection()
+        {
+            CarouselItem owner = this.OwnerCarouselItem;
+            Carousel parent = owner.ParentCarousel;
+            parent.SelectedItem = null;
+        }
+
+        /// <summary>Clears any existing selection and then selects the current element.</summary>
+        public void Select()
+        {
+            CarouselItem owner = this.OwnerCarouselItem;
+            Carousel parent = owner.ParentCarousel;
+            parent.SetSelectedItem(owner);
+        }
+
+        /// <summary>
+        /// Gets the control type for the element that is associated with the UI Automation peer.
+        /// </summary>
+        /// <returns>The control type.</returns>
+        protected override AutomationControlType GetAutomationControlTypeCore()
+        {
+            return AutomationControlType.ListItem;
+        }
+
+        /// <summary>
+        /// Called by GetClassName that gets a human readable name that, in addition to AutomationControlType,
+        /// differentiates the control represented by this AutomationPeer.
+        /// </summary>
+        /// <returns>The string that contains the name.</returns>
+        protected override string GetClassNameCore()
+        {
+            return Owner.GetType().Name;
+        }
+
+        /// <summary>
+        /// Called by GetName.
+        /// </summary>
+        /// <returns>
+        /// Returns the first of these that is not null or empty:
+        /// - Value returned by the base implementation
+        /// - Name of the owning CarouselItem
+        /// - Carousel class name
+        /// </returns>
+        protected override string GetNameCore()
+        {
+            int? index = this.OwnerCarouselItem.ParentCarousel.GetCarouselItems().ToList().IndexOf(this.OwnerCarouselItem);
```

Comment:
Can't we use `IndexFromContainer` here at least?

Reply:
@michael-hawker thanks for the recommendation on this one. Simplified this down.